luatest/utils.lua

```diff
@@ -191,4 +191,26 @@ function utils.is_tarantool_binary(path)
     return path:find('^.*/tarantool[^/]*$') ~= nil
 end
 
+function utils.unpack_sparse_array(array, index)
+    -- Embed everything so no upvalues required on server.
+    index = index or 1
+
+    local len = 0
+    for k, _ in pairs(array) do
```

Comment:
Maybe it makes sense to improve the `utils.table_len()` function?

```lua
function utils.table_len(t)
    local counter = 0
    for _, _ in pairs(t) do
        counter = counter + 1
    end
    return counter
end
```

Reply:
I'm not sure.

The concept of "array" is rather unclear in Lua, even though there are some tools like `ipairs` and `unpack`. The concept of "sparse array" is even more unclear. Now `table_len` is expected to return "the number of actual values in the table". 
For `{nil, 2, 3, nil, 5}` it's 3, but it's sparse array length is 5. And there is no any way for us to tell whether the developer is treating some table value as a "sparse array" and expect its length to be 5 or as a "key-value map with number keys" and expects its length to be 3. (The `{[index_id] = index}` map for a space schema is an example of key-value map with number keys. I have one in `tarantool/crud`. It always has `[0]` key, it may look like `{[0] = index_0, [2] = index_2}` if some index was removed. I'm more likely to expect that `len(index_map)` would return a count of indexes, which is the result of current `table_len`.) Since there is no way for us to choose the approach, it must be a responsibility of the developer. We may introduce `table_len` with flag to choose the approach, but I think it's more confusing than having two different functions.

(Now "sparse array len" is not a separate function since I dump the util and it's really a bother to support upvalues. But I can restore it to be a separate function in current implementation, even though in may confuse future `:exec` developers.)